### DISCUSSION TITLE
FlxInputManager bug or something else

### DISCUSSION THREAD

**Discussion by bogdan-ov** [HaxeFlixel/flixel#2515] [Q&A]
I use `FlixelInputManager` to control character(I put all actions in **isolated static** class). All work great, but when I reset the state via `FlxG.resetState()` input manager broke and character doesn't move... How can I solve it?
Code:
**Gamepad.hx** (static class with actions)
```Haxe
class Gamepad {
    static public var right = new FlxActionDigital("right")
        .addKey(RIGHT, PRESSED)
        .addKey(D, PRESSED);
    static public var left = new FlxActionDigital("left")
        .addKey(LEFT, PRESSED)
        .addKey(A, PRESSED);
    static public var up = new FlxActionDigital("up")
        .addKey(UP, JUST_PRESSED)
        .addKey(W, JUST_PRESSED);
    static public var down = new FlxActionDigital("down")
        .addKey(DOWN, JUST_PRESSED)
        .addKey(S, JUST_PRESSED);

    static public var shoot = new FlxActionDigital("shoot")
        .addKey(ENTER, PRESSED);
    static public var jump = new FlxActionDigital("jump")
        .addKey(SPACE, JUST_PRESSED);
    static public var dash = new FlxActionDigital("dash")
        .addKey(SHIFT, JUST_PRESSED);

    static public function horizontal():Int {
        return MyMath.boolToInt(right.triggered) - MyMath.boolToInt(left.triggered);
    }

    public static function getActions():Array<FlxAction> {
        return [right, left, up, down, shoot, jump, dash];
    }
}
```
**PlayState.hx**
```Haxe
...
override public function create() {
	super.create();	
	FlxG.inputs.add(new FlxActionManager()).addActions(Gamepad.getActions());
	...
}
...
```

**Comment by RichardBray** [comment]
Personally, I've never used `FlixelInputManager` before. It would be good to know what you're using it for and if it's the best solution for your case. Also...you might need to use a singleton, but that will depend on your answer to the first question.

**Comment by Geokureli** [comment]
I believe FlxG.inputs gets cleared every time you change states. i could be wrong tho

**Reply by bogdan-ov** [reply]
I think so too. But I tried to initialize inputs in `create()` method of `FlxState` class. Doesn't work

**Reply by Geokureli** [reply]
perhaps @larsiusprime can give insight into how these were intended to be used?

**Reply by Geokureli** [reply]
I notice inputs are instantiated in the player class of [this demo](https://github.com/HaxeFlixel/flixel-demos/blob/dev/Input/FlxAction/source/Player.hx). not sure if that helps

**Reply by bogdan-ov** [reply]
It is from this demo that I learned about FlxInputManager. Thanks for feedback! I don't need any more help)
